### PR TITLE
fix: script destroy bug

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: packages/core/src/ComponentsManager.ts [hunk 1]
```diff
@@ -127,8 +128,12 @@ export class ComponentsManager {
     renderer._onUpdateIndex = -1;
   }
 
-  addDestroyComponent(component): void {
-    this._destroyComponents.push(component);
+  addDisableScript(component: Script): void {
+    this._disableScripts.push(component);
+  }
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/ComponentsManager.ts:133]: removeDisableScript

**Review by cptbtptpbcptdtptp — COMMENTED** [review]
是否可以理解主要是修改了 _onDisable 的逻辑。

File: packages/core/src/ComponentsManager.ts [hunk 2]
```diff
@@ -160,7 +167,7 @@ export class ComponentsManager {
     const elements = this._onLateUpdateScripts._elements;
     for (let i = this._onLateUpdateScripts.length - 1; i >= 0; --i) {
       const element = elements[i];
-      if (element._started) {
+      if (element._isValid() &&element._started) {
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/ComponentsManager.ts:170]: && element._started

File: packages/core/src/ComponentsManager.ts [hunk 3]
```diff
@@ -170,7 +177,7 @@ export class ComponentsManager {
     const elements = this._onPhysicsUpdateScripts._elements;
     for (let i = this._onPhysicsUpdateScripts.length - 1; i >= 0; --i) {
       const element = elements[i];
-      if (element._started) {
+      if (element._isValid() &&element._started) {
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/ComponentsManager.ts:180]: && element._started

File: packages/core/src/ComponentsManager.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (GuoLei1990) [reply packages/core/src/ComponentsManager.ts]: done

File: packages/core/src/ComponentsManager.ts (hunk 3 above)
Comment (GuoLei1990) [reply packages/core/src/ComponentsManager.ts]: done

File: packages/core/src/ComponentsManager.ts (hunk 2 above)
Comment (GuoLei1990) [reply packages/core/src/ComponentsManager.ts]: done

**Review by yangfengzzz — APPROVED** [review]
+1

**Review by yangfengzzz — APPROVED** [review]
+1

**Pull request merged**